kv/txn_coord_sender.go

```diff
@@ -426,8 +428,24 @@ func (tc *TxnCoordSender) sendOne(ctx context.Context, call proto.Call) {
 	var startNS int64
 	header := call.Args.Header()
 	trace := tracer.FromCtx(ctx)
-	// If this call is part of a transaction...
+	var id string // optional transaction ID
 	if header.Txn != nil {
+		// If this call is part of a transaction...
+		id = string(header.Txn.ID)
+		// Verify that if this Transaction is not read-only, we have it on
+		// file. If not, refuse writes - the client must have issued a write on
+		// another coordinator previously.
+		if header.Txn.Writing && proto.IsTransactionWrite(call.Args) {
```

Comment:
There's no misbehaving client ;)

What will happen down the road (with the RFC) is that a transaction record contains the list of intents. If the client lies and lays out some intents through another coordinator, they will be aborted upon resolve (because they're not in the Txn record). So a misbehaving client loses the data that they write through the coordinator that they don't send the EndTransaction to.
Currently, the intents are written, and can be resolved. Either way, the client doesn't really win anything.